### PR TITLE
feat (web-iconics): add RTK Query to fetch data

### PR DESCRIPTION
**Pull request by FevenBelay23**
full page
Add RTK Query to fetch the data on the page from JSON file

### CODE REVIEW

**Comment by Mekdeskebede** [comment]
rename the PR name just for consistency

File: Web/starter_project_iconics/pages/api/success-stories.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { SuccessStory } from '../../types/story/success-stories'
+import story from '../../data/story/success-stories.json'
+
```
Comment (Mekdeskebede) [inline Web/starter_project_iconics/pages/api/success-stories.ts:4]: try to remove unnecessary spaces

File: Web/starter_project_iconics/pages/api/success-stories.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export default function handler(
```
Comment (zerihunMoges) [inline Web/starter_project_iconics/pages/api/success-stories.ts:6]: if the function is responsible for handling a specific type of API request related to success stories, you could name it something like handleSuccessStoryRequest or successStoryHandler.

File: Web/starter_project_iconics/store/features/story/success-stories-api.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    reducerPath: 'storiesApi',
+    baseQuery: fetchBaseQuery({baseUrl: API_BASE_URL}),
+    endpoints: (builder) => ({
+        getStories: builder.query<SuccessStory[], void>({
+            query: () => '/success-stories',
+        }),
+    }),
+})
+
+export const { useGetStoriesQuery} = storyApi
```
Comment (TsehaynehGetaneh) [inline Web/starter_project_iconics/store/features/story/success-stories-api.ts:16]: I am not sure, but I don't think you need the features folder, just add the story folder inside the store

**Review by TsehaynehGetaneh — CHANGES_REQUESTED** [review]

**Review by Henok-Matheas — APPROVED** [review]
except the Unescessary spaces, LGTM.

**Review by Henok-Matheas — APPROVED** [review]

**Review by ruthgdb — APPROVED** [review]

**Review by zerihunMoges — APPROVED** [review]

**Review by Mekdeskebede — APPROVED** [review]

**Review by ruthgdb — APPROVED** [review]

**Review by TsehaynehGetaneh — APPROVED** [review]

**Pull request merged**